src/platform/backends/qemu/qemu_platform.h

```diff
@@ -42,7 +42,11 @@ class QemuPlatform
     virtual optional<IPAddress> get_ip_for(const std::string& hw_addr) = 0;
     virtual void remove_resources_for(const std::string&) = 0;
     virtual void platform_health_check() = 0;
-    virtual QStringList platform_args(const VirtualMachineDescription& vm_desc)
+    virtual QStringList base_platform_args()
+    {
+        return {};
+    };
+    virtual QStringList full_platform_args(const VirtualMachineDescription& vm_desc)
```

Comment:
As discussed, IMO we should be explicit about how the args returned are used. So have specific `vmstate_`, `start_` methods here, and anything common between them should be handled in the platform-specific bit.

Reply:
I've renamed these `vmstate_platform_args()` and `vm_platform_args()`.